src/core/window.extern.js

```diff
@@ -22,5 +22,141 @@
+/* @public {string} */
+AmpConfigType.prototype.thirdPartyFrameHost;
+/* @public {string} */
+AmpConfigType.prototype.thirdPartyFrameRegex;
+/* @public {string} */
+AmpConfigType.prototype.errorReporting;
+/* @public {string} */
+AmpConfigType.prototype.betaErrorReporting;
+/* @public {string} */
+AmpConfigType.prototype.cdn;
+/* @public {string} */
+AmpConfigType.prototype.cdnUrl;
+/* @public {string} */
+AmpConfigType.prototype.errorReportingUrl;
+/* @public {string} */
+AmpConfigType.prototype.betaErrorReportingUrl;
+/* @public {string} */
+AmpConfigType.prototype.localDev;
+/* @public {string} */
+AmpConfigType.prototype.v;
+/* @public {boolean} */
+AmpConfigType.prototype.canary;
+/* @public {string} */
+AmpConfigType.prototype.runtime;
+/* @public {boolean} */
+AmpConfigType.prototype.test;
+/* @public {string|undefined} */
+AmpConfigType.prototype.spt;
+/* @public {boolean|undefined} */
+AmpConfigType.prototype.esm;
+/* @public {string} */
+AmpConfigType.prototype.geoApi;
+/* @public {string} */
+AmpConfigType.prototype.geoApiUrl;
+
+/** @type {boolean} */
+var IS_ESM;
+
+/** @type {string|undefined} */
+window.origin;
+
+/** @type {function(*, !Element=)|undefined} */
 window.__AMP_REPORT_ERROR;
+
+window.testLocation;
+window.Location.originalHash;
+window.__AMP_SERVICES;
+window.__AMP_TEST;
+window.__AMP_TEST_IFRAME;
+window.__AMP_TAG;
+window.__AMP_TOP;
+window.__AMP_PARENT;
+window.__AMP_WEAKREF_ID;
+window.__AMP_URL_CACHE;
+
+/** @type {?} */
+window.AMP = {};
+// window.AMP._ = {};
+window.AMP.push;
+window.AMP.title;
+window.AMP.canonicalUrl;
+window.AMP.extension;
+window.AMP.ampdoc;
+window.AMP.config;
+window.AMP.config.urls;
+window.AMP.BaseElement;
+window.AMP.registerElement;
+window.AMP.registerTemplate;
+window.AMP.registerServiceForDoc;
+window.AMP.isExperimentOn;
+window.AMP.toggleExperiment;
+window.AMP.setLogLevel;
+window.AMP.setTickFunction;
+window.AMP.viewer;
+
+/** @type {?} */
+window.AMP.viewport = {};
+window.AMP.viewport.getScrollLeft;
+window.AMP.viewport.getScrollWidth;
+window.AMP.viewport.getWidth;
+
+// Fullscreen methods
+Document.prototype.cancelFullScreen = function () {};
+Document.prototype.webkitExitFullscreen = function () {};
+Element.prototype.cancelFullScreen = function () {};
+Element.prototype.exitFullscreen = function () {};
+Element.prototype.webkitExitFullscreen = function () {};
+Element.prototype.webkitCancelFullScreen = function () {};
+Element.prototype.mozCancelFullScreen = function () {};
+Element.prototype.msExitFullscreen = function () {};
+Element.prototype.requestFullscreen = function () {};
+Element.prototype.requestFullScreen = function () {};
+Element.prototype.webkitRequestFullscreen = function () {};
+Element.prototype.webkitEnterFullscreen = function () {};
+Element.prototype.msRequestFullscreen = function () {};
+Element.prototype.mozRequestFullScreen = function () {};
+/** @type {boolean|undefined} */
+Element.prototype.webkitDisplayingFullscreen;
```

Comment:
I'd suggest element and document externs live elsewhere

Reply:
Reverted this change